### PR TITLE
Feature/cal and control

### PR DESCRIPTION
**Pull request by A-Daneel**
C&C pull request. calendar and commands.
did the tests as well. as far as what made sense.

I've noticed we return a lot of [datatype] | null, maybe we should rethink that...

### CODE REVIEW

File: tests/Unit/Calendar/ICalProviderTest.php [hunk 1]
```diff
@@ -15,32 +16,25 @@
...
-    /**
-     * @var ICalProvider
-     */
+    /** @var ICalProvider */
     protected $iCalProvider;
 
-    /**
-     * @var activity
-     */
+    /** @var activity */
```
Comment (jasperweyne) [inline tests/Unit/Calendar/ICalProviderTest.php:22]: Please replace this with `Activity` as this will improve typehints

**Review by jasperweyne — CHANGES_REQUESTED** [review]
A number of small comments. Please also take [this comment](https://github.com/jasperweyne/helpless-kiwi/pull/270/files#r838976231) on #270  into consideration, as those are also applicable here.

**Review by jasperweyne — CHANGES_REQUESTED** [review]
A small casings issue, but overall looks good

File: tests/Unit/Calendar/ICalProviderTest.php [hunk 2]
```diff
@@ -15,32 +16,25 @@
...
     protected $iCalProvider;
 
-    /**
-     * @var activity
-     */
+    /** @var Activity */
     protected $firstActivity;
+    /** @var activity */
     protected $secondActivity;
+    /** @var activity */
```
[the remark this answers is not in the document]
Comment (A-Daneel) [reply tests/Unit/Calendar/ICalProviderTest.php]: should be resolved now. first try success on the third attempt!

**Review by jasperweyne — APPROVED** [review]

**Pull request merged**